### ISSUE TITLE
About function: generate_anchor

### ISSUE THREAD

**Issue by professor1996** [foolwood/DaSiamRPN#15]
I cannot figure out how to generate anchor, in function generate_anchor() code:
""
    anchor = np.tile(anchor, score_size * score_size).reshape((-1, 4))
    ori = - (score_size / 2) * total_stride
    xx, yy = np.meshgrid([ori + total_stride * dx for dx in range(score_size)], [ori + total_stride * dy for dy in range(score_size)])
    xx, yy = np.tile(xx.flatten(), (anchor_num, 1)).flatten(), np.tile(yy.flatten(), (anchor_num, 1)).flatten()
    anchor[:, 0], anchor[:, 1] = xx.astype(np.float32), yy.astype(np.float32)
""
I print anchor. But output of anchor is very strange. Would you please explain these code?

My understanding is:
In feature map 17 * 17 * 256, therefore 289 anchors in total. Aera of every anchor's bbox is 64 * 64 in corresponding to ratios[0.33 , 0.5 , 1 , 2 , 3]. 
And mapped to  search image 255 * 255, each anchor generates 5 types of bbox(aera size: 64 * 64).
Stride is 255 / 17 = 16,  that is,  every 16 pixels exists one anchor,  uniformly distributing  in search image 255 * 255.
I got all regions of proposal.

Is that right?

**Comment by lb1100** [comment]
The stride must be the same as the network, is 8. The anchor position is in [255//2 - 17/2 * 8, 255//2 + 17/2 * 8], which is center at the original image center. **This is different from detection since our network do not have padding**, so some border part is 'ignored' by the network

**Comment by professor1996** [comment]
Thank you very much!!!  I have change my anchor position in stride:8
I came across another problem when training, I set batch_size in dataloader as 1,  the network in "SiamRPNBIG" worked and training loss dropped down .
But, if I set other batch_size( > 1) it appeals to wrong . It seems that, dimension of net mismatches with input image elements.
So could you give some suggestion to how to train SiamRPN net with larger patch_size?  Cuz it's not likely to use only 1 batch_size to train so big data...
